src/storage/redis_metadata.cc

```diff
@@ -495,3 +496,71 @@ rocksdb::Status HyperLogLogMetadata::Decode(Slice *input) {
 
   return rocksdb::Status::OK();
 }
+
+void CuckooFilterMetadata::Encode(std::string *dst) const {
+  Metadata::Encode(dst);
+  PutFixed64(dst, capacity);
+  PutFixed16(dst, bucket_size);
+  PutFixed16(dst, max_iterations);
+  PutFixed32(dst, num_buckets);
+  PutFixed16(dst, expansion);
+  PutFixed16(dst, num_filters);
+  PutFixed64(dst, num_items);
+  PutFixed64(dst, num_deletes);
+  for (const auto &filter : filters) {
+    PutFixed16(dst, filter.bucket_size);
+    PutFixed64(dst, filter.num_buckets);
+    dst->append(filter.data.begin(), filter.data.end());
```

Comment:
I doubt the efficiency of this approach: everytime we read/update the filters, we need to retrieve/put all of these filters.

Please refer to the design of bloom filters to see if you can put filters to multiple key values.